improc/photometry.py

```diff
@@ -204,17 +204,20 @@ def iterative_photometry(
         iterations = 0  # skip the iterative mode if there's no data
     else:
         # find a rough estimate of the centroid using non-tapered cutout
-        normalization = np.nansum(nandata)
-        if normalization < 1.0:
-            normalization = 1.0  # prevent division by zero and other rare cases
-        cx = np.nansum(xgrid * nandata) / normalization
-        cy = np.nansum(ygrid * nandata) / normalization
-        cxx = np.nansum((xgrid - cx) ** 2 * nandata) / normalization
-        cyy = np.nansum((ygrid - cy) ** 2 * nandata) / normalization
-        cxy = np.nansum((xgrid - cx) * (ygrid - cy) * nandata) / normalization
+        bkg_estimate = np.nanmedian(nandata)
+        normalization = np.nansum(nandata - bkg_estimate)
+        if normalization == 0:
+            normalization = 1.0
+        elif abs(normalization) < 1.0:
+            normalization = 1.0 * np.sign(normalization)  # prevent division by zero and other rare cases
+        cx = np.nansum(xgrid * (nandata - bkg_estimate)) / normalization
+        cy = np.nansum(ygrid * (nandata - bkg_estimate)) / normalization
+        cxx = np.nansum((xgrid - cx) ** 2 * (nandata - bkg_estimate)) / normalization
+        cyy = np.nansum((ygrid - cy) ** 2 * (nandata - bkg_estimate)) / normalization
+        cxy = np.nansum((xgrid - cx) * (ygrid - cy) * (nandata - bkg_estimate)) / normalization
```

Comment:
This is specifically for cutouts, and I am not sure in what situation you'd want to pass in the positions? You want to measure the position inside each cutout and pass it into the function? 

I am fine with calling this cutouts photometry. 

The second moments (`cxx`, `cyy`, `cxy`) from this step are used in case there's a failure and the function skips the iteration step (or if, for some strange reason, you give it `iterations=0`).